event_model/documents/stream_datum.py

```diff
@@ -18,40 +32,37 @@ class StreamDatum(TypedDict):
             "be contiguous for a given stream.",
         ),
     ]
-    datum_kwargs: Annotated[
-        Dict[str, Any],
+    stream_resource: Annotated[
+        str,
         Field(
-            description="Arguments to pass to the Handler to retrieve one "
-            "quanta of data",
+            description="The UID of the Stream Resource to which this Datum belongs."
         ),
     ]
-    event_count: Annotated[
-        int, Field(description="The number of events in this datum.")
-    ]
-    event_offset: Annotated[
-        int,
+    uid: Annotated[
+        str,
         Field(
-            description="The sequence number of the first event in this block. This "
-            "increasing value allows the presence of gaps.",
+            description="Globally unique identifier for this Datum. A suggested "
+            "formatting being '<stream_resource>/<stream_name>/<block_id>",
         ),
     ]
-    stream_name: Annotated[
-        str,
+    data_keys: Annotated[
+        List[str],
         Field(
-            description="The name of the stream that this Datum is providing a "
-            "block of.",
+            description="A list to show which data_keys of the "
+            "Descriptor are being streamed"
         ),
     ]
-    stream_resource: Annotated[
-        str,
+    seq_nums: Annotated[
+        StreamRange,
         Field(
-            description="The UID of the Stream Resource to which this Datum belongs."
+            description="A slice object showing the Event numbers the "
+            "resource corresponds to"
         ),
     ]
-    uid: Annotated[
-        str,
+    indices: Annotated[
+        StreamRange,
         Field(
-            description="Globally unique identifier for this Datum. A suggested "
-            "formatting being '<stream_resource>/<stream_name>/<block_id>",
+            description="A slice object passed to the StreamResource "
+            "handler so it can hand back data and timestamps"
         ),
```

Comment:
@evalott100 we also need a pointer back to the Descriptor here (copy what we have in `Event`)